### PR TITLE
PLNSRVCE-452: introduce in-repo e2e's against appstudio installed on openshift

### PR DESCRIPTION
**Pull request by gabemontero**

### CODE REVIEW

**Comment by gabemontero** [comment]
/hold

first vetting the script refactor

**Comment by gabemontero** [comment]
OK @stuartwdouglas @dwalluck @brunoapimentel @phillip-kruger @psturc  I have in repo e2e's running successfully against an openshift cluster with appstudio installed:

```
--- PASS: TestExampleRun (1023.08s)
    --- PASS: TestExampleRun/pipelinerun_completes_successfully (0.11s)
    --- PASS: TestExampleRun/artifactbuilds_and_dependencybuilds_generated (360.56s)
    --- PASS: TestExampleRun/taskruns_related_to_artifactbuilds_and_dependencybuilds_generated (0.54s)
    --- PASS: TestExampleRun/some_artfacdtbuilds_and_dependencybuilds_complete (660.85s)
PASS
ok  	github.com/redhat-appstudio/jvm-build-service/openshift-with-appstudio-test/e2e	1023.103s
gmontero ~/go/src/github.com/redhat-appstudio/jvm-build-service  (in-repo-e2e)$ 
```

these are the functional equivalent of the tests I've added in https://github.com/redhat-appstudio/e2e-tests/pull/142

Once this merges, I'll add the openshift/release changes to launch this e2e on our PRs, in addition to the existing e2e.  And of course we can all start adding to this suite, using it as a pre-merge vetting of new features and tests tests before adding new, after the fact, tests to redhat-appstudio/e2e-tests, which essentially amount to regression coverage after our jvm-build-service PRs merge.

File: openshift-with-appstudio-test/e2e/jvm_build_service_test.go
Comment (stuartwdouglas) [inline openshift-with-appstudio-test/e2e/jvm_build_service_test.go:173]: Not a blocker, but should we eventually use the tasks from the build definitions bundle here rather than this one?

**Review by stuartwdouglas — APPROVED** [review]

File: openshift-with-appstudio-test/e2e/jvm_build_service_test.go
Comment (gabemontero) [reply openshift-with-appstudio-test/e2e/jvm_build_service_test.go:173]: absolutely

that said, as you noted recently in https://docs.google.com/document/d/1qJvVaLWiWtkC4Gd2Q6ehSpxHBQLjLDxoGPdY9f_oYcY/edit?disco=AAAAcWLy7Ec we are not quite there yet, right?

Although I have not tried it super recently, that has been my experience as well so far when I try java-builder component builds with the currently hacbs recipes in build-definitions.

but as soon as that changes, this bit should pivot to use the build-definition stuff

**Pull request merged**
